discord/member.py

```diff
@@ -603,6 +603,31 @@ def guild_avatar(self) -> Asset | None:
             self._state, self.guild.id, self.id, self._avatar
         )
 
+    @property
+    def display_banner(self) -> Asset | None:
+        """Returns the member's display banner.
+
+        For regular members this is just their banner, but
+        if they have a guild specific banner then that
+        is returned instead.
+
+        .. versionadded:: 2.7
+        """
+        return self.guild_banner or self._user.banner
+
+    @property
+    def guild_banner(self) -> Asset | None:
+        """Returns an :class:`Asset` for the guild banner
+        the member has. If unavailable, ``None`` is returned.
+
+        .. versionadded:: 2.7
+        """
+        if self._user.banner is None:
+            return None
+        return Asset._from_guild_banner(
+            self._state, self.guild.id, self.id, self._user.banner
+        )
```

Comment:
```suggestion
        if self._banner is None:
            return None
        return Asset._from_guild_banner(
            self._state, self.guild.id, self.id, self._banner
        )
```

You should also add:
- to __init__
`self._banner: str | None = data.get("banner")`
- to _copy
`self._banner = member._banner`
- to _update
`self._banner = data.get("banner")`